altair/utils/save.py

```diff
@@ -9,28 +9,26 @@
 
 
 def write_file_or_filename(
-    fp: Union[str, pathlib.PurePath, IO],
+    fp: Union[pathlib.Path, IO],
```

Comment:
Let's keep `str` around as a type hint as the function actually works for strings as well. I don't think we need to remove this functionality. Also, in the Altair codebase there is unfortunately no clear separation between public and private functions/classes, i.e. most things need to be considered public. Removing `str` here could be a breaking change for someone who uses this function directly (although I'm not sure why anyone would).